Util/NativeHookUtil.cs

```diff
@@ -1,28 +1,48 @@
-
+using BepInEx.Unity.IL2CPP.Hook;
+using HarmonyLib;
+using MonoMod.RuntimeDetour;
 using System;
 using System.Reflection;
-using BepInEx.IL2CPP.Hook;
-using HarmonyLib;
+using System.Runtime.InteropServices;
 
 namespace Wetstone.Util;
 
 public static class NativeHookUtil
 {
-    public static FastNativeDetour Detour<T>(string ty, string methodName, T to, out T original) where T : System.Delegate?
+    public static NativeDetour Detour<T>(string typeName, string methodName, T to, out T original) where T : System.Delegate?
     {
-        return Detour(Type.GetType(ty), methodName, to, out original);
+        return Detour(Type.GetType(typeName), methodName, to, out original);
     }
 
-    public static FastNativeDetour Detour<T>(Type ty, string methodName, T to, out T original) where T : System.Delegate?
+    public static NativeDetour Detour<T>(Type type, string methodName, T to, out T original) where T : System.Delegate?
     {
-        var method = ty.GetMethod(methodName, AccessTools.all);
+        var method = type.GetMethod(methodName, AccessTools.all);
         return Detour(method, to, out original);
     }
 
-    public static FastNativeDetour Detour<T>(MethodInfo method, T to, out T original) where T : System.Delegate?
+    public static NativeDetour Detour<T>(MethodInfo method, T to, out T original) where T : System.Delegate?
+    {
+        var address = Il2CppMethodResolver.ResolveFromMethodInfo(method);
+        WetstonePlugin.Logger.LogInfo($"Detouring {method.DeclaringType?.FullName}.{method.Name} at {address.ToString("X")}");
+        var detour = new NativeDetour(address, Marshal.GetFunctionPointerForDelegate(to));
+        original = detour.GenerateTrampoline<T>();
+        return detour;
```

Comment:
What's the motivation behind providing an API around MonoMod's `NativeDetour` and BepInEx's `INativeDetour` at the same time?

You should only provide API that returns INativeDetour.